tikv/split_region.go

```diff
@@ -141,80 +142,57 @@ func (s *KVStore) batchSendSingleRegion(bo *Backoffer, batch kvrpc.Batch, scatte
 		}
 	}
 
-	req := tikvrpc.NewRequest(tikvrpc.CmdSplitRegion, &kvrpcpb.SplitRegionRequest{
-		SplitKeys: batch.Keys,
-	}, kvrpcpb.Context{
-		Priority: kvrpcpb.CommandPri_Normal,
-	})
+	opts := make([]pd.RegionsOption, 0, 1)
+	if tableID != nil {
+		opts = append(opts, pd.WithGroup(fmt.Sprintf("%v", *tableID)))
+	}
 
-	sender := locate.NewRegionRequestSender(s.regionCache, s.GetTiKVClient())
-	resp, err := sender.SendReq(bo, req, batch.RegionID, client.ReadTimeoutShort)
+	// Split regions by pd
+	spResp, err := s.pdClient.SplitRegions(bo.GetCtx(), batch.Keys, opts...)
 
-	batchResp := kvrpc.BatchResult{Response: resp}
-	if err != nil {
-		batchResp.Error = errors.Trace(err)
-		return batchResp
-	}
-	regionErr, err := resp.GetRegionError()
+	resp := SplitResponse{splitResp: spResp}
 	if err != nil {
-		batchResp.Error = errors.Trace(err)
-		return batchResp
-	}
-	if regionErr != nil {
-		err := bo.Backoff(retry.BoRegionMiss, errors.New(regionErr.String()))
-		if err != nil {
-			batchResp.Error = errors.Trace(err)
-			return batchResp
-		}
-		resp, err = s.splitBatchRegionsReq(bo, batch.Keys, scatter, tableID)
-		batchResp.Response = resp
-		batchResp.Error = err
-		return batchResp
+		resp.Error = err
+		return resp
 	}
 
-	spResp := resp.Resp.(*kvrpcpb.SplitRegionResponse)
-	regions := spResp.GetRegions()
-	if len(regions) > 0 {
+	regionsId := spResp.RegionsId
```

Comment:
It seems `integration_tests_test.(*testScanSuite).TestScan` panic due to nil pointer in this line.

Reply:
After we running the intergration env, we can run this testcase seprately by `go test -check.f testScanSuite.TestScan in its go package.

Reply:
I have solve this problem in pr : 
https://github.com/pingcap/tidb/pull/28336

Reply:
![image](https://user-images.githubusercontent.com/58988019/135709457-faa7af72-719c-4e37-8eaa-c750f098efb0.png)